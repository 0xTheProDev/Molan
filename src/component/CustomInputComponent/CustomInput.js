// @flow
import React from 'react';
import { Grid, Form, TextArea } from 'semantic-ui-react';
import { noop } from 'util/common';

import { CustomInputInternalProps } from './CustomInput.flow';

export function CustomInput({ theme, CustomInputPlaceholder, onCustomInputChange }: CustomInputInternalProps) {
  return (
    <Grid>
      <Grid.Row columns={1}>
        <Grid.Column stretched>
          <Form>
            <TextArea
              autoHeight
              placeholder={CustomInputPlaceholder}
              onChange={onCustomInputChange}
              className={theme}
            />
          </Form>
        </Grid.Column>
      </Grid.Row>
    </Grid>
  );
}

CustomInput.defaultProps = {
  CustomInputPlaceholder: 'Enter your input here',
  onCustomInputChange: noop,
};
