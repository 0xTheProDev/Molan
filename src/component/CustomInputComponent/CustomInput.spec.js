import React from 'react';
import { TextArea } from 'semantic-ui-react';

import { CustomInput } from './CustomInput';

describe('Component::CustomInput::CustomInput', () => {
  it('should render with base structure', () => {
    const wrapper = shallow(<CustomInput theme='light' />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke onChangeHandler when Input changes', () => {
    const onCustomInputChangeStub = jest.fn();

    const wrapper = shallow(<CustomInput theme='light' onCustomInputChange={onCustomInputChangeStub} />);
    wrapper
      .find(TextArea)
      .simulate('change');

    expect(onCustomInputChangeStub).toHaveBeenCalled();
  });
});
