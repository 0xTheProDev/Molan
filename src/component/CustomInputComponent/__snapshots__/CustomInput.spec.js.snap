// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component::CustomInput::CustomInput should render with base structure 1`] = `
<Grid>
  <GridRow
    columns={1}
  >
    <GridColumn
      stretched={true}
    >
      <Form
        as="form"
      >
        <TextArea
          as="textarea"
          autoHeight={true}
          className="light"
          onChange={[Function]}
          placeholder="Enter your input here"
          rows={3}
        />
      </Form>
    </GridColumn>
  </GridRow>
</Grid>
`;
